src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -217,20 +217,33 @@ private bool IsDiagnosticLocallySuppressed(string id, ISymbol symbol, out Suppre
             return suppressions.TryGetValue(id, out info);
         }
 
-        private ISymbol SuppressMessageAttribute
+        private ISymbol? SuppressMessageAttribute
         {
             get
             {
-                if (_lazySuppressMessageAttribute == null)
+                if (!_lazySuppressMessageAttribute.HasValue)
                 {
-                    _lazySuppressMessageAttribute = _compilation.GetTypeByMetadataName("System.Diagnostics.CodeAnalysis.SuppressMessageAttribute");
+                    _lazySuppressMessageAttribute = new(_compilation.GetTypeByMetadataName("System.Diagnostics.CodeAnalysis.SuppressMessageAttribute"));
                 }
 
-                return _lazySuppressMessageAttribute;
+                return _lazySuppressMessageAttribute.Value;
             }
         }
 
-        private void DecodeGlobalSuppressMessageAttributes()
+        private ISymbol? UnconditionalSuppressMessageAttribute
+        {
+            get
+            {
+                if (!_lazyUnconditionalSuppressMessageAttribute.HasValue)
+                {
+                    _lazyUnconditionalSuppressMessageAttribute = new(_compilation.GetTypeByMetadataName("System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute"));
```

Comment:
>_lazyUnconditionalSuppressMessageAttribute [](start = 20, length = 42)

Is this assignment thread-safe?

Reply:
I think so, in the specific way I was using it -- but rather than guess I just changed it to something I'm certain is safe.

Reply:
> I think so, in the specific way I was using it

It looks like Optional is a struct with several fields, the assignment is not guaranteed to be atomic. What am I missing?

---
In reply to: [597090036](https://github.com/dotnet/roslyn/pull/51792#discussion_r597090036) [](ancestors = 597090036)

Reply:
You're right, I the previous version wasn't thread safe. The path would be:

Thread 1:
_field.HasValue is false
retrieve Type
_field.HasValue <- true
--- interrupted here

Thread 2 (continuing at interruption):
_field.HasValue is true
return value is null (not correct)